tools/scripts/generate_changelog.py

```diff
@@ -0,0 +1,67 @@
+"""
+Generate changelog for github release from the changelog.rst file
+Example of execution:
+
+python tools/scripts/generate_changelog.py -v 1.15.0 -f docs/changelog.rst
+
+It outputs the result to the stdout
+"""
+
+import argparse
+import re
+
+NEW_ASSETS_MSG = 'Added support for the following tokens'
+
+def generate_changelog(version: str, path: str) -> str:
+    change_regex = r'\*\ :(feature|bug):`(\d*|-)`\ (.+)'
+
+    changes = {'features': [], 'bugs': []}
+    github_link = '- #{number} {text}'
+    not_listed_change = '- {text}'
+
+    with open(path) as f:
+        lines = f.readlines()
+        start = False
+        for line in lines:
+            if version in line:
```

Comment:
I would have sworn that this was in the last review but can't find the comment.

This is error prone. You should not search for release like that since it would consider any line that so much as mentions the version as the start of the version. You know more information about how a start ofthe version should look like.

```
* :release:`1.16.1
```

So use that prefix.